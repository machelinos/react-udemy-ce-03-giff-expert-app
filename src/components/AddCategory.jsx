import { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ onChangeCategory }) => {
  const [inputValue, setInputValue] = useState('')

  const handleInputChange = ({ target }) => {
    setInputValue(target.value)
  }

  const handleSubmit = (event) => {
    event.preventDefault()
    if (inputValue.trim().length <= 2) return
    onChangeCategory((cats) => [...cats, inputValue])
    setInputValue('')
  }

  return (
    <form onSubmit={handleSubmit}>
      <input
        type="text"
        placeholder="Search gifs..."
        value={inputValue}
        onChange={handleInputChange}
      />
    </form>
  )
}

AddCategory.propTypes = {
  onChangeCategory: PropTypes.func.isRequired,
}
