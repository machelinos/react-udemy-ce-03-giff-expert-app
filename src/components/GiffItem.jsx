import PropTypes from 'prop-types'
export const GiffItem = ({ url, title }) => {
  return (
    <div className="card">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  )
}

GiffItem.propTypes = {
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
}
