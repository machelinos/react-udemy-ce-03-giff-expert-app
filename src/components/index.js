export * from './AddCategory'
export * from './GiffGrid'
export * from './GiffItem'
