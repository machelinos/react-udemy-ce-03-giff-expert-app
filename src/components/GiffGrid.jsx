import PropTypes from 'prop-types'
import { useEffect, useState } from 'react'
import { fetchImages } from '../helpers/fetchImages'
import { GiffItem } from './GiffItem'

export const GiffGrid = ({ category }) => {
  const [images, setImages] = useState([])

  useEffect(() => {
    async function getImages() {
      const gifs = await fetchImages(category)
      setImages(gifs)
    }

    getImages()
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [])

  return (
    <div>
      <h2>{category}</h2>

      <div className="card-grid">
        {images.map((img) => (
          <GiffItem key={img.id} {...img} />
        ))}
      </div>
    </div>
  )
}

GiffGrid.propTypes = {
  category: PropTypes.string.isRequired,
}
