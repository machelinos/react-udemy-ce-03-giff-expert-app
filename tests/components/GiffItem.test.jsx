import { screen, render } from '@testing-library/react'
import { GiffItem } from '../../src/components/GiffItem'

describe('<GiffItem /> tests', () => {
  const urlImg =
    'https://media0.giphy.com/media/3o6Zt9MapEhDVITFpC/giphy.gif?cid=001d3785bwobjphsdmhyek0m4pyv3ufzylysh5redeoj9fq0&ep=v1_gifs_search&rid=giphy.gif&ct=g'
  const title = 'Buffy'

  test('<GiffItem /> render shoould match the snapshot', () => {
    const { container } = render(<GiffItem url={urlImg} title={title} />)
    expect(container).toMatchSnapshot()
  })

  test('<GiffItem /> should show an img with the alt and url passed', () => {
    render(<GiffItem url={urlImg} title={title} />)
    const { src, alt } = screen.getByRole('img')
    expect(src).toBe(urlImg)
    expect(alt).toBe(title)
  })

  test('<GiffItem /> should include title somewhere', () => {
    render(<GiffItem url={urlImg} title={title} />)
    expect(screen.getByText(title)).toBeTruthy()
  })
})
