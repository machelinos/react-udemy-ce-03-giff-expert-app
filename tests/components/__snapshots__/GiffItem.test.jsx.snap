// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GiffItem /> tests <GiffItem /> render shoould match the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Buffy"
      src="https://media0.giphy.com/media/3o6Zt9MapEhDVITFpC/giphy.gif?cid=001d3785bwobjphsdmhyek0m4pyv3ufzylysh5redeoj9fq0&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      Buffy
    </p>
  </div>
</div>
`;
